Create 10 quadraticProbing.cpp

Juro221 opened the pull request:
Similarly as in 09 linearProbing.cpp we compute the position and if it is occupied we are looking for the next free position.
Time complexity : O(n)

JurajZitnansky committed:
Create 10 quadraticProbing.cpp

Similarly as in 09 linearProbing.cpp we compute the position and if it is occupied we are looking for the next free position.
Time complexity : O(n)

FazeelUsmani reviewed (changes requested):
Getting wrong answer for the following test case:

Input:
41
11
862 364 256 810 113 908 153 56 685 595 100

Its Correct output is:
-1 862 -1 -1 -1 -1 908 -1 -1 -1 256 -1 -1 -1 -1 56 -1 -1 100 -1 -1 595 -1 -1 -1 -1 -1 -1 -1 685 153 810 113 -1 -1 -1 364 -1 -1 -1 -1

And Your Code's output is:
0 862 0 0 0 0 908 0 0 0 256 0 0 0 0 56 0 0 100 0 0...

JurajZitnansky committed:
Update 10 quadraticProbing.cpp

Forgot to initialize vector<int> hash to -1s.

FazeelUsmani reviewed (changes requested):
**Getting wrong answer**
The first test case where your code failed:

Input:
41
11
862 364 256 810 113 908 153 56 685 595 100

Its Correct output is:
-1 862 -1 -1 -1 -1 908 -1 -1 -1 256 -1 -1 -1 -1 56 -1 -1 100 -1 -1 595 -1 -1 -1 -1 -1 -1 -1 685 153 810 113 -1 -1 -1 364 -1 -1 -1 -1

And Your Code's output is:
-1 862 -1 -1 -1 -1 908 -1 -1 -1 256 -1 -1 -1 -1 56 -1 -1 100 -1 -1 595 -1 -1 -1 -1 -1 -1 -1 685 153 810 113 -1 -1 -1 364 -1 -1 -1 -1 -1 862 -1 -1 -1 -1 908 -1 -1 -1 256 -1 -1 -1 -1 5...

JurajZitnansky committed:
Update 10 quadraticProbing.cpp

I didn't notice that we dont need to print anything we just need to correctly fill our vector hash. It should be okay now.

FazeelUsmani reviewed (changes requested):
**Failing for the below test case:**

Wrong Answer. !!!Wrong Answer

Possibly your code doesn't work correctly for multiple test-cases (TCs).

The first test case where your code failed:

Input:
67
9
142 645 423 186 467 475 695 6 444

Its Correct output is:
-1 -1 -1 -1 -1 -1 475 6 142 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 423 -1 -1 -1 695 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 645 444 -1 -1 -1 -1 -1 -1 -1 -1 186 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 467 -1

And Your Code's output is:
-1 -1 -1 -1 -1 -1 475 6 142 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 423 -1 -1 -1 695 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 645 -1 -1 -1 444 -1 -1 -1 -1 -1 186 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 467 -1


Test against custom input
  
                position=(position + (counter * counter)) % hashSize;

FazeelUsmani commented:
This PR is not updated for a while, closing it. Thanks!
